xmtp_mls/src/storage/encrypted_store/mod.rs

```diff
@@ -362,6 +363,39 @@ macro_rules! impl_fetch {
     };
 }
 
+#[macro_export]
+macro_rules! impl_fetch_list {
+    ($model:ty, $table:ident) => {
+        impl $crate::FetchList<$model>
+            for $crate::storage::encrypted_store::db_connection::DbConnection
+        {
+            fn fetch_list(&self) -> Result<Vec<$model>, $crate::StorageError> {
+                use $crate::storage::encrypted_store::schema::$table::dsl::*;
+                Ok(self.raw_query(|conn| $table.load::<$model>(conn))?)
+            }
+        }
+    };
+}
+
+#[macro_export]
+macro_rules! impl_fetch_list_with_key {
+    ($model:ty, $table:ident, $key:ty, $column:ident) => {
+        impl $crate::FetchListWithKey<$model>
+            for $crate::storage::encrypted_store::db_connection::DbConnection
+        {
+            type Key = $key;
+            fn fetch_list_with_key(
+                &self,
+                keys: &[Self::Key],
+            ) -> Result<Vec<$model>, $crate::StorageError> {
```

Comment:
What would you think of having this return `Vec<Option<$model>>` instead so that way we can zip the output with the input of this function?

Reply:
It would make processing and matching the output data with the input data much easier. It would simplify the logic/processing you would have to do in the caching PR that you have [here](https://github.com/xmtp/libxmtp/pull/1226/files#diff-3c0189371525b6aae3be7e5a7b5b5682204ef341fb2744fc876148c5ecab8642R379-R396) and would avoid having to collect the data into a hashmap.

Reply:
Then I need to do the mapping in the macro itself after getting the data from the DB, and then in the situation, we don't find any data in the DB we return a `[ None ]` but now it's an empty vec, so I think with that we don't gain that much in making the return value as `Vec<Option<model>, StorageError>`!
what do you thing @insipx?